src/main/java/com/airbnb/airpal/resources/ResultsPreviewResource.java

```diff
@@ -82,7 +84,9 @@ private Response getS3Preview(URI fileURI, int numLines) {
                 outputKey
         ).withRange(0, 100 * 1024);
         val object = s3Client.getObject(request);
-        try (val s3Reader = new CSVReader(new BufferedReader(new InputStreamReader(object.getObjectContent())))) {
+        ObjectMetadata objectMetadata = object.getObjectMetadata();
+        boolean gzip = "gzip".equalsIgnoreCase(objectMetadata.getContentEncoding());
+        try (val s3Reader = new CSVReader(new BufferedReader(new InputStreamReader(gzip ? new GZIPInputStream(object.getObjectContent()) : object.getObjectContent())))) {
```

Comment:
Please break this line up a bit, it's a bit too complicated as it stands. Simply moving the result of the ternary to the result of a multiline `if..else` would be sufficient.